rdr_service/genomic/genomic_job_components.py

```diff
@@ -3070,65 +3081,67 @@ def generate_and_transfer_manifest(self, manifest_type, genome_type, **kwargs):
                 logging.info(
                     f'Preparing manifest of type {manifest_type}...'
                     f'{self.manifest_def.destination_bucket}/{self.output_file_name}'
                 )
-                self._write_and_upload_manifest(source_data)
-
-            results = []
-            record_count = len(source_data)
-            for row in source_data:
-                member = self.member_dao.get_member_from_sample_id(row.sample_id, genome_type)
+                self._write_and_upload_manifest(current_list)
 
-                if member is None:
-                    raise NotFound(f"Cannot find genomic set member with sample ID {row.sample_id}")
-
-                if self.manifest_def.job_run_field is not None:
-                    results.append(
-                        self.member_dao.update_member_job_run_id(
-                            member,
-                            job_run_id=self.run_id,
-                            field=self.manifest_def.job_run_field
-                        )
+        else:
+            self.output_file_name = self.manifest_def.output_filename
+            # If the new manifest is a feedback manifest,
+            # it will have an input manifest
+            if "input_manifest" in kwargs.keys():
+                # AW2F manifest file name is based of of AW1
+                if manifest_type == GenomicManifestTypes.AW2F:
+                    new_name = kwargs['input_manifest'].filePath.split('/')[-1]
+                    new_name = new_name.replace('.csv', '_contamination.csv')
+                    self.output_file_name = self.manifest_def.output_filename.replace(
+                        "GC_AoU_DataType_PKG-YYMM-xxxxxx_contamination.csv",
+                        f"{new_name}"
                     )
-                # Handle Genomic States for manifests
-                if self.manifest_def.signal != "bypass":
-                    new_state = GenomicStateHandler.get_new_state(member.genomicWorkflowState,
-                                                                  signal=self.manifest_def.signal)
-
-                    if new_state is not None or new_state != member.genomicWorkflowState:
-                        self.member_dao.update_member_state(member, new_state)
+            logging.info(
+                f'Preparing manifest of type {manifest_type}...'
+                f'{self.manifest_def.destination_bucket}/{self.output_file_name}'
+            )
+            self._write_and_upload_manifest(source_data)
+
+        results = []
+        for row in source_data:
+            member = self.member_dao.get_member_from_sample_id(row.sample_id, genome_type)
+            if member is None:
+                raise NotFound(f"Cannot find genomic set member with sample ID {row.sample_id}")
+
+            if self.manifest_def.job_run_field is not None:
+                results.append(
+                    self.member_dao.update_member_job_run_id(
+                        member,
```

Comment:
Does this need to be `member.id` now?

Reply:
Haha, that is from another PR that proceeds this one, https://github.com/all-of-us/raw-data-repository/pull/2575, that hasn't been merged yet.

Reply:
Makes sense.